### PR TITLE
T16: Profile page

### LINKED ISSUE #16: User profile page
This is the user profile which contains their avatar, background photo, bio, etc. The user's posts will render the timeline post component used on the timeline. MVP requirements:

- User should see their profile details excluding pictures and follower/following counts (username, display name, join date, etc)
  - Hard code a banner and either hard code or exclude follower counts for now. We'll address these in a different ticket
- Component should have tabs to filter between all of the user's posts, their replies, and their liked posts

### PR DESCRIPTION
**Pull request by AnindaS13**
This PR contains the complete profile page UI. The media tab was removed and the follower/following counts have been hardcoded as discussed. All three tabs (tweets, replies, likes) are functional with the backend ticket branch.

### CODE REVIEW

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Can we remove `userProfileArrow` and `rightBound` since they are not doing anything?

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Since all of these fetches require the same parameter, is there any way for us to manipulate the query so that it returns all the required information in just one fetch?

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Even though we can use the `width` attribute directly on `Box`, I think we should create a separate `styles.parentBox` or something for the CSS.

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: This might be nit-picky but I think it's best standard to follow MUI even for `div`. Basically, where you write `div`, make it `Box`.

https://stackoverflow.com/questions/60451957/what-is-the-material-ui-box-component-for

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Similar to the previous comment about `Box`, let's make this an `IconButton`, provide the same styling.

I would also suggest an `onClick` where you are able to revert to the previous page. You will have to 
 ```suggestion
 //outside of the render component: 
 const navigate = useNavigate();
 
 //inside of the render component:
<IconButton style={styles.arrowBtn} onClick={() => navigate(-1)}>
```

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx:148]: Remove the in-line styling and place it within the `styles` object

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: I believe this can also be a `Box` component with a variant of `img`. 

Take a look at this: https://stackoverflow.com/questions/61263669/does-material-ui-have-an-image-component

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Move the in-line styling to `styles` object

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Move the in-line styling to `styles` object

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx:189]: Try to use `Typography` for this

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx:194]: Same comment as above

**Review by NasihNazeem — CHANGES_REQUESTED** [review]

File: src/components/Posts/ProfileLikes.tsx
Comment (NasihNazeem) [inline src/components/Posts/ProfileLikes.tsx:1]: I like the simplicity, and using the PostList as a template. Is there any way for us to combine `ProfileLikes`, `ProfileReplies` and `ProfileTweets` or do you think that would make the component too complex?

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: Good use of Tabs component!

File: src/pages/Profile.tsx
Comment (NasihNazeem) [inline src/pages/Profile.tsx]: What does the empty brackets do?

File: src/components/Posts/ProfileLikes.tsx
Comment (Dennull) [reply src/components/Posts/ProfileLikes.tsx:1]: I agree with this. The post list components are basically the same except for the endpoint they fetch data from. I commented this on #41, but we have the same issue in our reply list for an expanded post. I suggested on there that we could make a new ticket to sort that out

File: src/pages/Profile.tsx
Comment (Dennull) [reply src/pages/Profile.tsx]: Seconded, it's a nitpick but for consistency I think we should stick to keeping our css in the `styles` object unless there's a good reason for inlining it

File: src/pages/Profile.tsx
Comment (Dennull) [reply src/pages/Profile.tsx]: Yeah, was actually going to bring this up at a meeting, but we should talk about how we feel about using MUI's component over normal html elements. I don't have a strong opinion but it's probably better to be consistent in the codebase

I had similar question with MUI's `Stack` component, which is just a flexbox container. You can use a `div` or `Box` and put flex properties on it to achieve the same thing, so I got confused about which one I should use

File: src/pages/Profile.tsx
Comment (Dennull) [inline src/pages/Profile.tsx]: In terms of styling my preference is that the header has bigger font than the number of tweets. Let me know if anyone has a feels differently

File: src/pages/Profile.tsx
Comment (Dennull) [inline src/pages/Profile.tsx]: Can we please remove this log?

File: src/pages/Profile.tsx
Comment (Dennull) [reply src/pages/Profile.tsx]: One nitpick, I think since `handleChange` isn't being reused and its logic isn't overly complex, we can inline it as an arrow function. My suggestion is to do something like this instead:

```suggestion
          <Tabs value={value} onChange={(_: React.SyntheticEvent, newValue: string) => setValue(newValue)}>
```

I prefer inlining in this case because it's not a lot of code. If the onChange was a lot more code I'd agree with this approach to not bloat the jsx

File: src/pages/Profile.tsx
Comment (Dennull) [reply src/pages/Profile.tsx]: There are a few of these in this file, can we please remove them?

**Review by Dennull — CHANGES_REQUESTED** [review]

File: src/pages/Profile.tsx
Comment (Dennull) [inline src/pages/Profile.tsx]: Can we please rename these to be camel case? Same goes for the backend the returns them

File: src/pages/Profile.tsx
Comment (Dennull) [inline src/pages/Profile.tsx]: Let's remove this comment if we don't need it

File: src/pages/Profile.tsx
Comment (Dennull) [inline src/pages/Profile.tsx]: Can we please make components without children self closing?
```suggestion
              <CalendarMonthIcon/>
```

**Review by Dennull — APPROVED** [review]
I have a few nitpicks and ideas about how we can simplify some markup, but I don't think any of them are worth blocking the PR over. If you're down tho we can get on call one day before merging and talk about it!

File: src/pages/Profile.tsx
Comment (NasihNazeem) [reply src/pages/Profile.tsx:172]: agree

**Review by NasihNazeem — APPROVED** [review]

**Pull request merged**
